### ISSUE TITLE
Feature Request: Algolia-like aroundPrecision parameter

### ISSUE THREAD

**Issue by SaxJ** [typesense/typesense#321]
## Description

Geo-ranking is a popular requirement for search. I would be enormously useful for my use-case to have an `aroundPrecision` parameter that works in the same way as Algolia, so that ranking can fall through from geo-ranking to the next criteria.

### Reference
https://www.algolia.com/doc/api-reference/api-parameters/aroundPrecision/

PS: Thank you so much for everybody's work on this project!

**Comment by kishorenc** [comment]
👋 @SaxJ This recently did cross my mind. For now, we do have a similar configuration option that might be useful for a subset of use cases: https://typesense.org/docs/0.21.0/api/documents.html#sorting-by-additional-attributes-within-a-radius

Instead of bucketing all distances, which `aroundPrecision` seems to do, `exclude_radius` will apply to only records within an immediate `exclude_radius` value. Records within this radius have the "fall through" but records outside will not. 

We should certainly support bucketing across all distances, like `aroundPrecision` as well.

**Comment by SaxJ** [comment]
Thank you for your fast response. `exclude_radius` will indeed do nicely for now.

**Comment by kishorenc** [comment]
@SaxJ This is available in `typesense/typesense:0.22.0.rc6`

Like `exclude_radius` you can now set `geo_precision` (only one of the two can be set at any time).

Example:

```
'sort_by' : 'location(48.853, 2.344, geo_precision: 2mi):asc, popularity:desc'
```

This will bucket all points into `2 mile` slots so that the popularity metric can fall through. Please give it a spin and let me know if you have any feedback!

**Comment by jasonbosco** [comment]
Released in v0.22.
